keps/sig-instrumentation/4827-component-statusz/README.md

```diff
@@ -0,0 +1,424 @@
+* Streamlined troubleshooting and debugging:  Immediate access to crucial build information like binary version, Go version and compatibility details enables better understanding of the component's state. We hope this interface makes it easier to identify version mismatches or incompatibilities that might contribute to issues, accelerating troubleshooting and debugging efforts
+
+
+### Goals
+
+* Enhance observability into component details by providing a concise overview of critical component information, such as build version, Go version, compatibility details, and other pertinent data points
+* Set a precedent for presenting this basic but crucial component information in a consistent and accessible manner across the Kubernetes ecosystem
+
+### Non-Goals
+
+* Replace existing monitoring solutions: metrics, logs, traces
+* To provide a general-purpose extension mechanism for other components (e.g., container runtime, CNI, device plugins) to expose their own information through z-pages
+* Provide information about components that are not directly accessible due to network restrictions
+
+## Proposal
+
+We are proposing to add a new endpoint, /statusz on all core Kubernetes components which returns key details regarding a component's current state, including versioning, build information, and compatibility.
+
+### User Stories
+
+#### As a developer
+
+- I want to quickly identify the exact version of a binary running in production so I can correlate it with known issues or recent code changes
+- I want to quickly access the compatibility version information of a deployed binary so I can efficiently debug upgrade failures and identify potential compatibility issues with dependencies
+
+#### As an operator
+
+- I need to troubleshoot a performance issue and want to easily check for dependency conflicts or version mismatches without sifting through logs
+
+#### As support engineer
+
+- I want to access detailed build information to determine if a customer is running a known problematic version of a component
+
+### Risks and Mitigations
+
+1. **No sensitive data exposed**
+    
+  We will apply the  principle of least privilege by granting only the minimum necessary permissions for accessing /statusz, which is primarily a debugging tool for operators and maintainers.
+  /statusz will not be accessible through general discovery mechanisms like system:discovery. This prevents unintended exposure of potentially sensitive debugging information.
```

Comment:
Will the /statusz endpoint support [k8s auditing](https://kubernetes.io/docs/tasks/debug/debug-cluster/audit/) and are accesses to /statusz logged somewhere?  Wondering if there might be a way for users to monitor access to these endpoints while the feature is alpha to understand how potentially sensitive debugging information is being accessed.

Reply:
Users should be able to specify a policy specific to the nonResource url /statusz and may capture audit logs for it I think, unsure if we need to explicitly log access to statusz though..